support/cas-server-support-pac4j-core-clients/src/main/java/org/apereo/cas/support/pac4j/logout/TriggerCasSLOLogoutHandler.java

```diff
@@ -0,0 +1,92 @@
+import java.util.function.Predicate;
+
+/**
+ * The pac4j logout handler which triggers a CAS logout.
+ *
+ * @author Jerome LELEU
+ * @since 7.0.0
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class TriggerCasSLOLogoutHandler implements LogoutHandler {
+
+    private final CasCookieBuilder ticketGrantingTicketCookieGenerator;
+
+    private final TicketRegistry ticketRegistry;
+
+    private final ConfigurableApplicationContext applicationContext;
+
+    @Override
+    public void recordSession(final WebContext context, final SessionStore sessionStore, final String key) {
+        context.setRequestAttribute(DelegatedTicketGrantingTicketFactory.DELEGATED_SESSION_KEY_REQUEST_ATTRIBUTE, key);
+    }
+
+    @Override
+    public void destroySessionFront(final WebContext context, final SessionStore sessionStore,
+                                    final ProfileManagerFactory profileManagerFactory, final String key) {
+        triggerCasSLO(context, sessionStore, key);
+    }
+
+    @Override
+    public void destroySessionBack(final WebContext context, final SessionStore sessionStore,
+                                   final ProfileManagerFactory profileManagerFactory, final String key) {
+        triggerCasSLO(context, sessionStore, key);
+    }
+
+    protected void triggerCasSLO(final WebContext context, final SessionStore sessionStore, final String key) {
+        val requestContext = RequestContextHolder.getRequestContext();
+        if (requestContext != null) {
+            val externalContext = requestContext.getExternalContext();
+            if (externalContext instanceof ServletExternalContext sec) {
+                val request = (HttpServletRequest) sec.getNativeRequest();
+                var tgtId = ticketGrantingTicketCookieGenerator.retrieveCookieValue(request);
+                LOGGER.debug("Found TGT cookie value: [{}]", tgtId);
+                if (StringUtils.isBlank(tgtId)) {
+                    val predicate = (Predicate<Ticket>) ticket -> ticket instanceof TicketGrantingTicketImpl
+                                                        && !ticket.isExpired()
+                                                        && StringUtils.equals(((TicketGrantingTicketImpl) ticket).getLinkedIdentifier(), key);
+                    val optTicket = ticketRegistry.getTickets(predicate).findFirst();
+                    LOGGER.debug("Found TGT: [{}] for key: [{}]", optTicket, key);
+                    if (optTicket.isPresent()) {
+                        tgtId = optTicket.get().getId();
+                    }
+                }
+
+                if (StringUtils.isNotBlank(tgtId)) {
+                    LOGGER.debug("Performing CAS SLO for tgt: [{}]", tgtId);
+                    try {
+                        WebUtils.putTicketGrantingTicketInScopes(requestContext, tgtId);
+                        val action = (Action) applicationContext.getBean(CasWebflowConstants.ACTION_ID_TERMINATE_SESSION);
```

Comment:
To finally call the `TerminateSessionAction`: I prefer to use the action to remove the cookies compared to the `LogoutManager`.